File: hash_table/hand_of_straights.py
class Solution:
    def isNStraightHand(self, hand, W):
        count = collections.Counter(hand)
        for key in sorted(c.keys()):
            if count[key] > 0:
                for j in range(W)[::-1]:
                    count[key + j] -= count[key]
                    if count[key + j] < 0:
                        return False
        return True

File: hash_table/hand_of_straights.java
class Solution {
    public boolean isNStraightHand(int[] hand, int W) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        for(int i : hand){
            minHeap.add(i);
        }
        while(minHeap.size() != 0) {
            int start = minHeap.poll();
            for(int j = 1; j < W; j++){
                if(minHeap.remove(start + j)) {
                    continue;
                }
                else {
                    return false;
                }
            }
        }
        return true;
    }
}